sensu/plugins/metrics-network-queues.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+#
+#  metrics-network-queues.py
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#  Python 2.7+ (untested on Python3, should work though)
+#
+# USAGE:
+#
+# metrics-network-queues.py -n <process_name> [-s <graphite_scheme>]
+#
+# DESCRIPTION:
+# Finds the pid[s] corresponding to a process name and obtains the lengths of
+# the receive and send network queues for all of the process' sockets. This
+# data is gathered from the "netstat -tpane" command.
+#
+# Code adapted from Jaime Gogo's script in the Sensu Plugins community
+# https://github.com/sensu-plugins/sensu-plugins-process-checks/blob/master/bin/metrics-per-process.py
+#
+# Released under the same terms as Sensu (the MIT license); see MITLICENSE
+# for details.
+#
+# Siva Mullapudi <scmullap@us.ibm.com>
+
+import optparse
+import sys
+import os
+import time
+import subprocess
+from collections import Counter
+
+STATE_OK = 0
+STATE_CRITICAL = 2
+
+PROC_ROOT_DIR = '/proc/'
+
+def find_pids_from_name(process_name):
+    '''Find process PID from name using /proc/<pids>/comm'''
+
+    pids_in_proc = [ pid for pid in os.listdir(PROC_ROOT_DIR) if pid.isdigit() ]
+    pids = []
+    for pid in pids_in_proc:
+        path = PROC_ROOT_DIR + pid
+        if 'comm' in os.listdir(path):
+            file_handler = open(path + '/comm', 'r')
+            if file_handler.read().rstrip() == process_name:
+                pids.append(int(pid))
+    return pids
+
+def queue_lengths_per_pid(pid):
+    '''Gets network rx/tx queue lengths for a specific pid'''
+
+    process_queues = {'receive_queue_length': 0, 'send_queue_length': 0}
+    netstat = subprocess.Popen(('netstat', '-tpane'), stdout=subprocess.PIPE)
+
+    try:
+        process_sockets = subprocess.check_output \
+                          (('grep', '%d' % pid), stdin=netstat.stdout)
```

Comment:
Uh, Python has better methods to find a substring within a string than shelling out to grep.

I'd recommend getting the output of netstat into a variable, and then using the native python tooling to inspect that output.